### PR TITLE
Task #41395 "Add Another Item" Clears Current Selection on BFN Form

### PR DESCRIPTION
**Pull request by snapik**
Added overlay to bulk node form

### CODE REVIEW

File: bulk_photo_nodes.module [hunk 1]
```diff
@@ -23,12 +54,21 @@ function bulk_photo_nodes_permission() {
...
-      $items[$node_add_path]['access arguments'] = array('create bulk photo nodes');
+
+    foreach (entity_get_info() as $entity_type => $entity_info) {
+      if ($entity_type !== 'node') {
+        continue;
+      }
+
+      foreach (array_keys($entity_info['bundles']) as $bundle) {
+        if (isset($bpn_var[$bundle]) && !empty($bpn_var[$bundle . '_override']) && (!empty($bpn_var[$bundle]) || $bpn_var[$bundle] !== 'none')) {
+          $node_add_path = 'node/add/' . strtr($bundle, '_', '-');
```
Comment (generalredneck) [inline bulk_photo_nodes.module:63]: I woudl consider making this a function. You use it at least twice

File: bulk_photo_nodes.module [hunk 2]
```diff
@@ -117,6 +173,15 @@ function bulk_photo_nodes_get_image_field($node_type) {
   return 'none';
 }
 
+function bulk_photo_nodes_get_option($node_type, $option = 'override') {
+  $bpn_var = variable_get('bulk_photo_node_types');
+  if (is_array($bpn_var)) {
+    $key = $node_type . '_' . $option;
+    return (array_key_exists($key, $bpn_var)) ? $bpn_var[$key] : 'none';
+  }
```
Comment (generalredneck) [inline bulk_photo_nodes.module:178]: Consider reversing your logic. Typically it's better to not have the majority of your logic in an If statement.

In this case 

```
if (!is_array($bpn_var)) {
  return 'none';
}
$key = $node_type . '_' .$option;
return (array_key_exists($key, $bpn_var)) ? $bpn_var[$key] : 'none';
```

This puts the emphasis on the last 2 lines as the main sting.

File: bulk_photo_nodes.module [hunk 3]
```diff
@@ -132,45 +197,46 @@ function bulk_photo_nodes_get_image_field($node_type) {
  * @see bulk_photo_nodes_menu_alter()
  */
 function bulk_photo_nodes_page($node_type) {
-
   if (bulk_photo_nodes_is_bpn_add_page($node_type)) {
+    return bulk_photo_nodes_page_get($node_type);
+  }
+  else {
```
Comment (generalredneck) [inline bulk_photo_nodes.module:201]: Seems you don't need the else here...

File: bulk_photo_nodes.module [hunk 4]
```diff
@@ -337,78 +449,62 @@ function bulk_photo_nodes_add_form($form, &$form_state) {
...
-  );
+/**
+ * Insert batch saved files into bulk_photo_nodes_add_form state.
+ */
+function bulk_photo_nodes_get_session_files(&$form_state) {
+  if (!empty($_SESSION['saved_files'])) {
+    $form_state['saved_files'] = $_SESSION['saved_files'];
+  }
+  if (empty($form_state['saved_files'])) {
+    drupal_goto("node/add/$form_state[node_type]");
```
Comment (generalredneck) [inline bulk_photo_nodes.module]: You did a replacement in the menu items with _ and -. Is that needed here? see https://github.com/promet/bulk_file_nodes/pull/1/files#diff-ab3838fd2f7dad40332648c2b2065c3eR65

File: bulk_photo_nodes.module (hunk 1 above)
Comment (snapik) [inline bulk_photo_nodes.module:63]: Added new function here

File: bulk_photo_nodes.module (hunk 3 above)
Comment (snapik) [inline bulk_photo_nodes.module:201]: Looks like we need this else if we don't have a BFN form here

**Comment by snapik** [comment]
FYI: All those changes made by @kid-icarus So it isn't all my code =) 
My changes only related to .js and .css files

**Pull request merged**